yaptide/batch/batch_methods.py

```diff
@@ -327,3 +327,31 @@ def delete_job(simulation: BatchSimulationModel, user: KeycloakUserModel,
         return {"message": "Job cancelation failed"}, 500
 
     return {"message": "Job canceled"}, 200
+
+
+def cancel_simulation_with_fetching_data(simulation: BatchSimulationModel, user: KeycloakUserModel,
+                                         cluster: ClusterModel) -> tuple[dict, int]:  # skipcq: PYL-W0613
+    """Function that stops simulation without cancelling collection job"""
+    array_id = simulation.array_id
+    tasks = fetch_batch_tasks_by_sim_id(sim_id=simulation.id)
+    try:
+        con = get_connection(user=user, cluster=cluster)
+        i = 1
```

Comment:
why do you need `i` here ?

Reply:
I do not see i being incremented...

Reply:
It is used in a loop under, but i don't remember to add a incrementation of this in that loop.

Reply:
maybe an `for i, task in enumerate(tasks)` would do the job as well ?